src/dvc_objects/fs/implementations/gdrive.py

```diff
@@ -155,45 +155,48 @@ def _validate_credentials(auth, settings):
                     "`gdrive_user_credentials_file` for one or more of them?"
                 )
 
-    def _try_load_env_creds(self):
-        import json
-
-        data = os.getenv(self.GDRIVE_CREDENTIALS_DATA)
-        if not data:
-            return None
-
-        try:
-            return json.loads(data)
-        except ValueError as exc:
-            raise GDriveAuthError(self.GDRIVE_CREDENTIALS_DATA) from exc
-
     @wrap_prop(threading.RLock())
     @cached_property
     def fs(self):
+        import json
+
         from pydrive2.auth import GoogleAuth
         from pydrive2.fs import GDriveFileSystem as _GDriveFileSystem
 
         auth_settings = {
             **GoogleAuth.DEFAULT_SETTINGS,
             "client_config_backend": "settings",
             "save_credentials": True,
-            "save_credentials_backend": "file",
-            "save_credentials_file": self._gdrive_user_credentials_path,
             "get_refresh_token": True,
             "oauth_scope": [
                 "https://www.googleapis.com/auth/drive",
                 "https://www.googleapis.com/auth/drive.appdata",
             ],
         }
 
+        env_creds_json = os.getenv(self.GDRIVE_CREDENTIALS_DATA)
+        try:
+            env_creds_dict = json.loads(env_creds_json)
```

Comment:
pydrive and oauth API have a bit of a mismatch and some take dict but not json and vise-versa, so we have to use both for now. Will add helper to pydrive2 later.